source/mod_particles.f90

```diff
@@ -20,55 +18,133 @@ end subroutine part_allocate
 subroutine part_expand
 end subroutine part_expand
 
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Last modified: 2018-10-31
+!  Moved from maincr. Applies the closed orbit correction to the particle coordinates.
+! ================================================================================================ !
+subroutine part_applyClosedOrbit
+
+  use mod_common
+  use mod_commons
+  use mod_commonmn
+
+  implicit none
+
+  integer j
+
+  if(iclo6 == 2) then
+    xv1(1:napx)   = xv1(1:napx)   +  clo6v(1,1:napx)
+    yv1(1:napx)   = yv1(1:napx)   + clop6v(1,1:napx)
+    xv2(1:napx)   = xv2(1:napx)   +  clo6v(2,1:napx)
+    yv2(1:napx)   = yv2(1:napx)   + clop6v(2,1:napx)
+    sigmv(1:napx) = sigmv(1:napx) +  clo6v(3,1:napx)
+    dpsv(1:napx)  = dpsv(1:napx)  + clop6v(3,1:napx)
+  else if(idfor == 0) then
+    xv1(1:napx)   = xv1(1:napx)   +   clov(1,1:napx) * real(idz(1),fPrec)
+    yv1(1:napx)   = yv1(1:napx)   +  clopv(1,1:napx) * real(idz(1),fPrec)
+    xv2(1:napx)   = xv2(1:napx)   +   clov(2,1:napx) * real(idz(2),fPrec)
+    yv2(1:napx)   = yv2(1:napx)   +  clopv(2,1:napx) * real(idz(2),fPrec)
+  end if
+  call part_updatePartEnergy(3)
+
```

Comment:
not clear to me why the `part_applyClosedOrbit` anyway calls `part_updatePartEnergy`, i.e. why if the user decides to add the closed orbit, no matter if 4D or 6D the long variables are anyway re-computed. I think we should call `part_updatePartEnergy` outside `part_applyClosedOrbit`.

If I correctly remember, beams from both `fort.13` and `DIST` block will never add the closed orbit, right?

Reply:
I agree, but this is how it was done before. Removing this will break a large number of tests.

Reply:
ok - let's wait for such a cleanup once we are able to re-validate all tests following code modifications which entangle changes in the last digits of results... do you agree, @rdemaria  ?